wings/market/coinbase_pro_market.pyx

```diff
@@ -0,0 +1,816 @@
+                                                                     tracked_order.quote_asset,
+                                                                     (tracked_order.fee_asset
+                                                                      or tracked_order.quote_asset),
+                                                                     float(tracked_order.executed_amount),
+                                                                     float(tracked_order.quote_asset_amount),
+                                                                     float(tracked_order.fee_paid)))
+                else:
+                    self.logger().info(f"The market order {tracked_order.client_order_id} has failed according to "
+                                       f"order status API.")
+                    self.c_trigger_event(self.MARKET_TRANSACTION_FAILURE_EVENT_TAG,
+                                         MarketTransactionFailureEvent(
+                                             self._current_timestamp,
+                                             tracked_order.client_order_id
+                                         ))
+                self.c_stop_tracking_order(tracked_order.client_order_id)
+
+    async def place_order(self, order_id: str, symbol: str, amount: Decimal, is_buy: bool, order_type: OrderType,
+                          price: float):
+        path_url = "/orders"
+        data = {
+            "price": price,
+            "size": float(amount),
+            "product_id": symbol,
+            "side": "buy" if is_buy else "sell",
+            "type": "limit" if order_type is OrderType.LIMIT else "market",
+        }
+
+        order_result = await self._api_request("post", path_url=path_url, data=data)
+        return order_result
+
+    async def execute_buy(self,
+                          order_id: str,
+                          symbol: str,
+                          amount: float,
+                          order_type: OrderType,
+                          price: Optional[float] = None):
+        cdef:
+            TradingRule trading_rule = self._trading_rules[symbol]
+
+        decimal_amount = self.quantize_order_amount(symbol, amount)
+        if decimal_amount < trading_rule.base_min_size:
+            raise ValueError(f"Buy order amount {decimal_amount} is lower than the minimum order size "
+                             f"{trading_rule.base_min_size}.")
+
+        try:
+            self.c_start_tracking_order(order_id, "", symbol, True, order_type, decimal_amount, price)
+            order_result = await self.place_order(order_id, symbol, decimal_amount, True, order_type, price)
+            self.c_trigger_event(self.MARKET_BUY_ORDER_CREATED_EVENT_TAG,
+                                 BuyOrderCreatedEvent(self._current_timestamp, order_type, symbol, decimal_amount,
+                                                      0.0 if price is None else price, order_id))
```

Comment:
suggestion - maybe leave price as None, seems misleading to change it to 0.0

Reply:
im assuming that price is only None for market orders

Reply:
ok